datastore/gcloud/rest/datastore/key.py

```diff
@@ -0,0 +1,90 @@
+from typing import Any  # pylint: disable=unused-import
+from typing import Dict  # pylint: disable=unused-import
+from typing import List  # pylint: disable=unused-import
+from typing import Optional  # pylint: disable=unused-import
+
+
+class PathElement(object):
+    def __init__(self, kind, **kwargs):
```

Comment:
Could we make `id_` and `name` explicit like we have in `gcloud-aio` rather than using `kwargs`?

Reply:
The thing is, we can't make them keyword-only. And only one of them can be passed here.

One alternative is to force the caller to pass `None, "name"` when they want to use a name.
Another is to pass an `id_or_name` and look at the data type.

I'll think about it more.

Reply:
I think I'm missing something -- why can we not make this `def __init__(self, kind, id_=None, name=None):` in the same way as we do in `aio`? I believe `aio` has a check in the body of the init method (`if id_ and name: raise ...`)

Reply:
Nah I'm the one missing something :)
Some habits are hard to break and it didn't occur to me that I could use the method signature you propose and still call it with `PathElement(Kind, name='name')` - I blame my C# days